tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -8613,3 +8613,192 @@ TEST_F(EnergyPlusFixture, SurfaceGeometry_GetKivaFoundationTest2)
+        "    ,                        !- View Factor to Ground",
+        "    4,                       !- Number of Vertices",
+        "    0,0,0,              !- Vertex 1",
+        "    0,1,0,              !- Vertex 2",
+        "    1,1,0,              !- Vertex 3",
+        "    1,0,0;              !- Vertex 4",
+
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+    bool ErrorsFound = false;
+
+    GetMaterialData(*state, ErrorsFound); // read material data
+    EXPECT_FALSE(ErrorsFound);            // expect no errors
+
+    GetConstructData(*state, ErrorsFound); // read construction data
+    EXPECT_FALSE(ErrorsFound);             // expect no errors
+
+    GetZoneData(*state, ErrorsFound); // read zone data
+    EXPECT_FALSE(ErrorsFound);        // expect no errors
+
+    SetupZoneGeometry(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound); // expect no errors
+
+    // Zone 1 with user-entered floor area of 30.0 consists of Space 1 and Space 2
+    // Space 1 has a floor surface of area 1.0, user-entered floor area is blank
+    // Space 2 has a floor surface of area 2.0, user-entered floor area is blank
+    EXPECT_EQ(state->dataHeatBal->space(1).Name, "SPACE 1A");
+    EXPECT_NEAR(state->dataHeatBal->space(1).userEnteredFloorArea, DataGlobalConstants::AutoCalculate, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(1).calcFloorArea, 1.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(1).floorArea, 10.0, 0.001);
+
+    EXPECT_EQ(state->dataHeatBal->space(2).Name, "SPACE 1B");
+    EXPECT_NEAR(state->dataHeatBal->space(2).userEnteredFloorArea, DataGlobalConstants::AutoCalculate, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(2).calcFloorArea, 2.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(2).floorArea, 20.0, 0.001);
+
+    EXPECT_EQ(state->dataHeatBal->Zone(1).Name, "ZONE 1");
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).UserEnteredFloorArea, 30.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).CalcFloorArea, 3.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).FloorArea, 30.0, 0.001);
+    Real64 zone1Area = state->dataHeatBal->space(1).floorArea + state->dataHeatBal->space(2).floorArea;
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).FloorArea, zone1Area, 0.001);
+
+    // Zone 3 consists of Space 3, user-entered floor area is blank
+    // Space 3 has a floor surface of area 1.0, user-entered floor is 5.0
+    EXPECT_EQ(state->dataHeatBal->Zone(3).Name, "ZONE 3");
+    EXPECT_NEAR(state->dataHeatBal->Zone(3).UserEnteredFloorArea, DataGlobalConstants::AutoCalculate, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->Zone(3).CalcFloorArea, 5.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->Zone(3).FloorArea, 5.0, 0.001);
+    EXPECT_EQ(state->dataHeatBal->space(3).Name, "SPACE 3");
+    EXPECT_NEAR(state->dataHeatBal->space(3).userEnteredFloorArea, 5.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(3).calcFloorArea, 1.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->space(3).floorArea, 5.0, 0.001);
```

Comment:
Zone 3 reversed which floor area was blank compared to zone 1 and areas still match